api/v1beta1/cluster_types.go

```diff
@@ -40,6 +40,236 @@ const (
+const (
+	// ClusterScalingDownV1Beta2Condition is true if replicas > desired replicas.
+	ClusterScalingDownV1Beta2Condition = ScalingDownV1Beta2Condition
+
+	// ClusterScalingDownV1Beta2Reason surfaces when at least one of the Cluster's control plane, MachineDeployments,
+	//	// MachinePools and stand-alone MachineSets are scaling down.
+	ClusterScalingDownV1Beta2Reason = ScalingDownV1Beta2Reason
+
+	// ClusterNotScalingDownV1Beta2Reason surfaces when no one of the Cluster's control plane, MachineDeployments,
+	// MachinePools and stand-alone MachineSets are scaling down.
+	ClusterNotScalingDownV1Beta2Reason = NotScalingUpV1Beta2Reason
+
+	// ClusterScalingDownInternalErrorV1Beta2Reason surfaces unexpected failures when listing machines
+	// or computing the ScalingDown condition.
+	ClusterScalingDownInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+)
+
+// Cluster's Remediating condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterRemediatingV1Beta2Condition surfaces details about ongoing remediation of the controlled machines, if any.
+	ClusterRemediatingV1Beta2Condition = RemediatingV1Beta2Condition
+
+	// ClusterRemediatingV1Beta2Reason surfaces when the Cluster has at least one machine with HealthCheckSucceeded set to false
+	// and with the OwnerRemediated condition set to false.
+	ClusterRemediatingV1Beta2Reason = RemediatingV1Beta2Reason
+
+	// ClusterNotRemediatingV1Beta2Reason surfaces when the Cluster does not have any machine with HealthCheckSucceeded set to false
+	// and with the OwnerRemediated condition set to false.
+	ClusterNotRemediatingV1Beta2Reason = NotRemediatingV1Beta2Reason
+
+	// ClusterRemediatingInternalErrorV1Beta2Reason surfaces unexpected failures when computing the Remediating condition.
+	ClusterRemediatingInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+)
+
+// Cluster's Deleting condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterDeletingV1Beta2Condition surfaces details about ongoing deletion of the cluster.
+	ClusterDeletingV1Beta2Condition = DeletingV1Beta2Condition
+
+	// ClusterDeletingDeletionTimestampNotSetV1Beta2Reason surfaces when the Cluster is not deleting because the
+	// DeletionTimestamp is not set.
+	ClusterDeletingDeletionTimestampNotSetV1Beta2Reason = DeletionTimestampNotSetV1Beta2Reason
+
+	// ClusterDeletingWaitingForBeforeClusterDeleteHookV1Beta2Reason surfaces when the Cluster deletion
+	// waits for the ClusterDelete hooks to allow deletion to complete.
+	ClusterDeletingWaitingForBeforeClusterDeleteHookV1Beta2Reason = "WaitingForClusterDeleteHook"
+
+	// ClusterDeletingWaitingForMachineDeletionV1Beta2Reason surfaces when the Cluster deletion
+	// waits for the Machines and the object controlling machines (MachinePools, MachineDeployments, MachineSets)
+	// to be deleted.
+	ClusterDeletingWaitingForMachineDeletionV1Beta2Reason = "WaitingForMachineDeletion"
```

Comment:
Fixed

Caveat: in case folks are using stand-alone CP machines they are also deleted in this phase, but I think what is proposed works very well in the most common use cases.

Reply:
Sounds good!